### ISSUE TITLE
SignalR sometimes throws "undefined is not a function" after reconnect

### ISSUE THREAD

**Issue by antross** [SignalR/SignalR#3362]
I've been periodically seeing the following error after SignalR passes through the `Reconnecting` state:

```
TypeError: undefined is not a function
    at Object.i.transports._logic.tryInitialize (jquery.signalR-2.1.2.min.js:8:18411)
    at Object.i.transports._logic.processMessages (jquery.signalR-2.1.2.min.js:8:18767)
    at EventSource.<anonymous> (jquery.signalR-2.1.2.min.js:8:24203)
```

I dug into the referenced SignalR code and noticed the following:

1. When reconnecting, SignalR [calls `transport.start`](https://github.com/SignalR/SignalR/blob/master/src/Microsoft.AspNet.SignalR.Client.JS/jquery.signalR.transports.common.js#L541) without passing a success or failure callback. 
2. Then the undefined callback gets [passed to `processMessages`](https://github.com/SignalR/SignalR/blob/master/src/Microsoft.AspNet.SignalR.Client.JS/jquery.signalR.transports.webSockets.js#L118) when `socket.onmessage` fires in the `WebSocket` transport.
3. Next `processMessages` [passes the undefined callback to `tryInitialize`](https://github.com/SignalR/SignalR/blob/master/src/Microsoft.AspNet.SignalR.Client.JS/jquery.signalR.transports.common.js#L424).
4. Finally `tryInitialize` [blindly invokes the undefined callback](https://github.com/SignalR/SignalR/blob/master/src/Microsoft.AspNet.SignalR.Client.JS/jquery.signalR.transports.common.js#L394).

Unless I missed a check somewhere, this appears to be the root cause of the error I'm seeing. At a minimum `tryInitialize` should probably check to ensure the callback exists before invoking it, though perhaps a callback should be provided at a higher level.

**Comment by niemyjski** [comment]
I just got this reported to us by our JavaScript Error client as well:

```js
TypeError: b is not a function
at Object.i.transports._logic.tryInitialize in https://be.exceptionless.io/app.min.1f50f6bcfa24aa04.js:line 22:col 26920
at Object.i.transports._logic.processMessages in https://be.exceptionless.io/app.min.1f50f6bcfa24aa04.js:line 22:col 27276
at EventSource.<anonymous> in https://be.exceptionless.io/app.min.1f50f6bcfa24aa04.js:line 23:col 773
```

**Comment by MaceWindu** [comment]
Seen this error 4 times already with the same stack:

    Reported Error: TypeError: onInitialized is not a function

    Reported Error Stack: signalR.transports._logic.tryInitialize
    signalR.transports._logic.processMessages
    signalR.transports.serverSentEvents.start/<

As @antross said before, it's due to reconnect do not pass function parameter to start() and tryInitialize calling it without check.
Browsers: 2 errors from FF32 and 2 errors from FF35

**Comment by moozzyk** [comment]
Fixed in 2e7308fc67114551123ec63123582a986ff5aad4 and 96e8261f527e8d6a89ff7366e14dccd67f0adb74
